src/ctap/crypto_wrapper.rs

```diff
@@ -89,11 +101,37 @@ pub fn aes256_cbc_decrypt(
 }
 
 /// An asymmetric private key that can sign messages.
-#[derive(Clone, Debug, PartialEq, Eq)]
+#[derive(Debug)]
 pub enum PrivateKey {
     Ecdsa(ecdsa::SecKey),
+    #[cfg(feature = "with_ed25519")]
+    Ed25519(ed25519_dalek::Keypair),
+}
+
+impl Clone for PrivateKey {
+    fn clone(&self) -> Self {
+        match self {
+            Self::Ecdsa(sk) => Self::Ecdsa (sk.clone ()),
+            #[cfg(feature = "with_ed25519")]
+            Self::Ed25519(keypair) => Self::Ed25519 (ed25519_dalek::Keypair::from_bytes (&keypair.to_bytes()).unwrap()),
```

Comment:
See my high level PR comment. Doesn't seem great.

Reply:
I agree. This is not maintainable.